Framework/DataObjects/src/EventList.cpp

```diff
@@ -3868,11 +3866,12 @@ void EventList::filterByPulseTime(Kernel::TimeROI *timeRoi, EventList &output) c
  *  The splitter intervals must be sorted by start time, which is not enforced
  *  in this helper function and is expected to be implicitly followed by the
  *  caller.
- *
- *  Also, a future version using TimeROI will automatically sort the intervals.
  */
-template <class T>
-void EventList::filterInPlaceHelper(Kernel::SplittingIntervalVec &splitter, typename std::vector<T> &events) {
+template <class T> void EventList::filterInPlaceHelper(Kernel::TimeROI *timeRoi, typename std::vector<T> &events) {
+  if (timeRoi == nullptr) {
```

Comment:
A second check should be for `timeRoi->useAll()` and throwing an exception with a different message. 

Since `filterInPlaceHelper` is private, maybe these checks should be moved up to the public `filterInPlace`. I do not hold this opinion strongly so feel free to ignore it.

Reply:
I added the other check and moved them to the public `filterInPlace` that made more sense to me.